File: carcassonne-master/src/right_side.cc
#include "right_side.h"

#include "right_side_connections.h"

namespace carcassonne
{

RightSide::RightSide() : Side()
{
	middle_.reset(new TerrainSegment);
	connections_.reset(new RightSideConnections);
}

RightSide::RightSide(const RightSide& src) : Side(src)
{
	CopyFrom(src);	
}

RightSide& RightSide::
	operator=(const RightSide& rhs)
{
	if(this == &rhs) {
		return (*this);
	}
	
	CopyFrom(rhs);
	
	return (*this);
}
 	
void RightSide::
	CopyFrom(const RightSide& src)
{
}
 
Side* RightSide::
	Clone() const
{
	return new RightSide(*this);
}

void RightSide::ConnectedToTop()
{
	connections_->set_counterclockwise(true);
	
}

void RightSide::ConnectedToRight()
{}

void RightSide::ConnectedToBottom()
{
	connections_->set_clockwise(true);
	
}

void RightSide::ConnectedToLeft()
{
	connections_->set_across(true);
	
}

bool RightSide::IsConnectedToTop()
{
  return connections_->counterclockwise();
}

bool RightSide::IsConnectedToRight()
{
  return true;
}

bool RightSide::IsConnectedToBottom()
{
  return connections_->clockwise();
}

bool RightSide::IsConnectedToLeft()
{
  return connections_->across();
}

std::string RightSide::ToString() const
{
  return "right side with " + connections_->ToString();
  
}

RightSide::~RightSide()
{
}

}

File: js/RightSide.js
/**
 * Created by marcogobbi on 10/10/2017.
 */
const Side=require( "./Side");
const RightSideConnections=require( "./RightSideConnections");
const TerrainSegment=require( "./terrain/TerrainSegment");
module.exports= class RightSide extends Side {
    constructor(src) {
        super(src);
        this.middle_ = new TerrainSegment();
        this.connections_ = new RightSideConnections();
    }

    Clone() {
        return new RightSide(this);
    }

    ConnectedToTop() {
        this.connections_.set_counterclockwise(true);

    }

    ConnectedToRight() {


    }

    ConnectedToBottom() {
        this.connections_.set_clockwise(true);

    }

    ConnectedToLeft() {this.connections_.set_across(true);
    }

    IsConnectedToTop() {
        return this.connections_.counterclockwise();

    }

    IsConnectedToRight() {
        return true;

    }

    IsConnectedToBottom() {
        return this.connections_.clockwise();

    }

    IsConnectedToLeft() {
        return this.connections_.across();

    }

    ToString() {
        return "right side with " + this.connections_.ToString();

    }
}